src/_posts/databases/postgresql/2000-01-01-upgrading.md

```diff
@@ -0,0 +1,233 @@
+---
+title: Upgrading Your Scalingo for PostgreSQL® Addon
+nav: Upgrading
+modified_at: 2024-11-27 12:00:00
+tags: databases postgresql addon
+index: 6
+---
+
+In Scalingo terminology, ***upgrading*** a Scalingo for PostgreSQL® addon
+designates the operation consisting in changing the database version for a
+newer one.
+
+We disinguish two main cases:
+- a ***minor-upgrade*** designates an incremental release with a change in the
+  tenths digit of the version number (e.g. `13.3.4` to `13.5.0`). It usually
+  provides bugs or security fixes as well as minor additional features.
+- a ***major-upgrade*** designates an incremental release with a change in the
+  main version number (e.g. `13.x.y` to `14.0.0`). Major-upgrades often provide
+  significant changes and new features that **can** break backward
+  compatibility.
+
+While we usually advise to stick to the latest minor-upgrade available to
+benefit from bug and security fixes, we also **strongly** advise to take
+extra-care when major-upgrading your Scalingo for PostgreSQL® addon ([more
+about this below](#best-practices-when-managing-major-upgrades)).
+
+When the database vendor releases a new version of your database engine, we
+take some time to study it and test it thoroughly before making it available.
+Upgrading to this new version is still your choice. We never do it
+automatically.
+
+{% warning %}
+Beware that no downgrade is possible once your database has been upgraded.
+{% endwarning %}
+
+## Understanding the Minor-Upgrade Process
+
+### Prerequisites
+
+There are no prerequisites for minor-upgrades.
+
+### Process
+
+#### For Starter Plans
+
+1. The instance is stopped. The database is unreachable.
+2. We restart the instance with the version targeted.
+3. Once the instance restarted, the database is reachable again.
+4. The application is restarted to ensure proper connexions. This shouldn't
+   cause any additional downtime.
+
+Since we have to completely stop the instance, **a downtime is inevitable**.
+
+We usually roughly estimate the downtime caused by the operation to a few
+seconds (2 to 10 seconds), depending on the platform load.
```

Comment:
question: doesn't it depend on the amount of data in the database?

Reply:
Apparently not since we don't run `pg_upgrade` when dealing with minor-upgrades.
The volume containing data is "attached" to the new container and that's it.

Reply:
Yes but from what I remember, when PostgreSQL starts it needs to read again the data on the disk and the duration of this operation depends on the amount of data.

Reply:
Do you mean we also run the `ANALYZE` SQL command after a minor-ugrade?
Is the database available or not during this read?

Reply:
No we don't execute `ANALYZE` for a minor upgrade.

>  Is the database available or not during this read?

No.

PostgreSQL needs to read the data that is on the disk and load them into the RAM. That's what is taking time.

Reply:
Fixed in #2885